docs/source/infer.autoguide.rst

```diff
@@ -125,6 +125,38 @@ AutoGaussian
     :member-order: bysource
     :show-inheritance:
 
+AutoMessenger
+-------------
+.. autoclass:: pyro.infer.autoguide.AutoMessenger
+    :members:
+    :undoc-members:
+    :member-order: bysource
+    :show-inheritance:
+
+AutoNormalMessenger
+-------------------
+.. autoclass:: pyro.infer.autoguide.AutoNormalMessenger
+    :members:
+    :undoc-members:
+    :member-order: bysource
+    :show-inheritance:
+
+AutoHierarchicalNormal
```

Comment:
Could you call this `AutoHierarchicalMessenger` or other `Auto*Messenger` to distinguish it from other `Auto*` guides?  I think it's important to add `Messenger` to the name because all these `GuideMessenger` guides require different inference machinery (`Effect_ELBO` rather than `Trace_ELBO`) and it's easiest to remind users "if it ends with `Messenger`, use `Effect_ELBO`"?  Sorry for flip flopping!

Reply:
Ok, that makes sense too. I renamed it `AutoHierarchicalNormalMessenger` because it is a fully descriptive name.